src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -661,6 +662,20 @@ def _prep_net_for_train(self) -> None:
         self.resume_from_remote_sg_logger = core_utils.get_param(self.training_params, "resume_from_remote_sg_logger", False)
         self.load_checkpoint = self.load_checkpoint or self.external_checkpoint_path is not None or self.resume_from_remote_sg_logger
 
+        if self.training_params.torch_compile:
+            if torch_version_is_greater_or_equal(2, 0):
+                logger.info("Using torch.compile feature. Compiling model. This may take a few minutes")
+                self.net = torch.compile(self.net, **self.training_params.torch_compile_options)
+                logger.info("Model compilation complete. Continuing training")
+                if is_distributed():
+                    torch.distributed.barrier()
+            else:
+                logger.warning(
+                    "Your recipe has requested use of torch.compile. "
+                    "However torch.compile is not supported in this version of PyTorch. "
+                    "Ignoring torch_compile flag"
+                )
```

Comment:
Good logs, but I would also print the required version in the warning to let the user know how to update and actually use torch.compile

Reply:
Fixed in 4b584f5db0809b12fa422fd93a33a38bcaa3d735